clusterloader2/pkg/provider/provider.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
```

Comment:
Meta comment - It would simplify the review, if you could split the PR into two parts:
- introduce pkg/provider (the interface and individual implementation) - this PR
- start using that (i.e. migration to use it) in the second PR

Reply:
Done, there is another PR: 1358